apiserver/facades/client/charms/services/downloader.go

```diff
@@ -68,3 +68,25 @@ func (s repoFactoryShim) GetCharmRepository(ctx context.Context, src corecharm.S
 
 	return repo, err
 }
+
+type loggoLoggerFactory struct {
+	Logger loggo.Logger
```

Comment:
I would embed `charmhub.LoggerFactory` here instead of duplicating `Child` and `ChildWithLabels`.

Reply:
I'll see what I can do with this one, it was painful because of invariance the last time I tried.

Reply:
Bitten by invariance.